### PR TITLE
[WIP] Integrate Graylog with rocksteady

### PR DESCRIPTION
**Pull request by MarcoBgn**
@jbilbo I don't have access to PowerRhino, feel free to add me if needed. 

The flow we discussed has been implemented here. I didn't go into too much refactoring as you might want to change things. Please feel free to, as many choices are still fairly opinionated.

### Context

The feature is fully retro-compatible an transparent to all users without: `GRAYLOG_ENABLED`  

- From the UI the checkbox is not visible and the feature is skipped entirely. 
- From an API the `add_graylog_stream` parameter (optional) is used instead

### CODE REVIEW

File: app/services/graylog_api/manager.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+module GraylogApi
+  class Manager
+    ROLE = 'Dev'.freeze
```
Comment (jbilbo) [inline app/services/graylog_api/manager.rb:3]: We can leave `Dev` as the default value, but shall we include the Role name as another dynamic attribute?

**Comment by octopusinvitro** [comment]
The Graylog API tests were completely rewritten to omit not only tests that were documenting relevant behaviour, but also edge cases. The current tests just check that the result is a success or a failure, leaving parts of the production code untested and undocumented. :thinking: I'm afraid it is a step backwards rather than an improvement. 

I thought maybe [the original code](https://github.com/digital-science/altmetric-utilities/pull/206) was changed too, so that is why the tests needed to change. However I can see the production code was ported mostly with no changes. :shrug:

File: app/services/graylog_api/client.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
+require 'http'
+
+module GraylogApi
```
Comment (octopusinvitro) [inline app/services/graylog_api/client.rb:3]: The Ruby style guide states that acronyms like API should be left uppercased:

https://github.com/rubocop-hq/ruby-style-guide#camelcase-for-classes

File: app/models/app_deletion.rb [hunk 3]
```diff
@@ -15,4 +17,8 @@ def delete!
   def url
     ENV.fetch('NOMAD_API_URI') + '/v1/job/' + app.name
   end
+
+  def delete_graylog_stream
+    GraylogApi::Manager.new(app).delete_stream
```
Comment (octopusinvitro) [inline app/models/app_deletion.rb:22]: shall we keep all the knowledge about streams encapsulated in the app
so instead of calling this class here, we would just do `app.delete_stream`
and the app would know who to call

File: app/services/graylog_api/manager.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      role.update(stream.id)
+
+      {
+        result: result,
+        stream_id: stream.id
+      }
+    end
+
+    def delete_stream
+      client.delete(Stream::ENDPOINT, app.graylog_stream.stream_id)
```
Comment (octopusinvitro) [inline app/services/graylog_api/manager.rb:28]: The knowledge about how to delete a stream should live in the stream class.

**Comment by MarcoBgn** [comment]
Thanks for the comments @jbilbo @octopusinvitro . As this was meant to be a hand-over branch will close it and incorporate the feedback in a new one
